_posts/2025-05-07-release-notes-3.7.0.md

```diff
@@ -0,0 +1,458 @@
+```
+
+Scala 3.7 provides an automated migration path for existing codebases through the compiler flags -rewrite -source:3.7-migration, which automatically inserts the recommended `using` keywords, streamlining the transition to the new syntax.
+
+### Scala 3 unblocked on Android [#22632](https://github.com/scala/scala3/pull/22632)
+
+Scala 3.7 brings a crucial fix that enhances its compatibility with the Android platform. Previously, developers faced issues when compiling Scala 3 code for Android due to the Android Runtime (ART) enforcing stricter type constraints on lambda expressions compared to the standard Java Virtual Machine (JVM). Specifically, ART requires that the return type of a Single Abstract Method (SAM) interface be a primitive type or explicitly boxed, a condition not mandated by the JVM.​
+
+The update addresses this by modifying the Scala compiler to box the return type of native instantiated methods when the SAM's return type isn't primitive. This change ensures that lambda expressions conform to ART's expectations, preventing runtime errors on Android devices. By aligning the compiler's behaviour with ART's requirements, the Scala 3 development for the Android platform should be unblocked, although it might require recompiling existing libraries using Scala 3.7 or the upcoming Scala 3.3 LTS version containing a backported fix.
+
+### Minimal support for dependent case classes [#21698](https://github.com/scala/scala3/pull/21698)
+
+Starting with Scala 3.7, developers can now define case classes with dependent fields, marking a notable improvement in expressiveness and type safety. This allows fields within a case class to depend on other constructor parameters via path-dependent types. One use case is encoding a configuration system where each setting has a distinct value type determined by the setting itself.
+
+```scala
+case class ConfigEntry(option: Setting, default: option.Value):
+  def value(using Context): option.Value = option.userDefined.getOrElse(default)
+
+trait Context
+trait Setting(val name: String):
+  type Value
+  def userDefined(using Context): Option[Value] = None
+
+class StringSetting(name: String) extends Setting(name):
+  override type Value = String
+
+class IntSetting(name: String) extends Setting(name):
+  override type Value = Int
+
+
+@main def DependentCaseClasses =
+  given Context = new {}
+  val settings = List[ConfigEntry](
+    ConfigEntry(StringSetting("docs.page"), "https://docs.scala-lang.org/"),
+    ConfigEntry(IntSetting("max.threads"), 32)
+  )
+```
+
+This enhancement simplifies type-safe heterogeneous collections and makes it easier to express advanced design patterns previously requiring workarounds or more verbose type definitions.
+
+### Dependency updates
+
+#### Scala 2 Standard Library
+
+Scala 3 now defaults to the [Scala 2.13.16](https://github.com/scala/scala/releases/tag/v2.13.16) Standard Library. Be mindful of breaking changes and deprecations introduced since 2.13.15:
```

Comment:
```suggestion
Scala 3 now uses the [Scala 2.13.16](https://github.com/scala/scala/releases/tag/v2.13.16) standard library. Be mindful of these breaking changes and deprecations since 2.13.15:
```